### PR TITLE
Harden unit tests

### PR DESCRIPTION
**Pull request by JcolemanNR**
Makes unit tests less likely to fail locally, and in CI

### CODE REVIEW

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs [hunk 1]
```diff
@@ -717,6 +720,8 @@ public void ErrorsWithTheRequestStream_ShouldShutdownResponseStream()
 
         [Test]
         [Ignore("This test is flickering in our CI", Until = "2022-06-01 00:00:00Z")]
```
Comment (nr-ahemsath) [inline tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:722]: If we're confident this test will now pass reliably, let's remove the Ignore attribute.

Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:722]: They have passed twice in CI now. I'll remove this and see if we can get a 3rd pass :)

**Review by michaelgoin — APPROVED** [review]

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs [hunk 2]
```diff
@@ -965,6 +971,8 @@ public void SupportabilityMetrics_Errors_OnSend()
...
```
[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs]: Ctrl-F5 and all your wildest dreams will come true!

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs [hunk 3]
```diff
@@ -750,13 +755,14 @@ public void ShuttingDownTheDataStreamingService_ShouldShutdownResponseStream()
...
-            NrAssert.Multiple
-            (
-                () => Assert.IsTrue(signalIsDone.Wait(TimeSpan.FromSeconds(10)), "Signal didn't fire"),
-                () => Assert.AreEqual(2, streamCancellationTokens.Count, "Did not see enough streams created"),
-                () => Assert.IsTrue(streamCancellationTokens[0].IsCancellationRequested, "The first stream cancellation token was not triggered."),
-                () => Assert.IsTrue(streamCancellationTokens[1].IsCancellationRequested, "The second stream cancellation token was not triggered.")
-            );
+
+            Assert.IsTrue(signalIsDone.Wait(TimeSpan.FromSeconds(10)), "Signal didn't fire");
+            Thread.Sleep(TimeSpan.FromSeconds(5));
```
[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:759]: In general, there are multiple span consumers, and the signal gets set when the first span consumer is disposed, not the last

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs [hunk 4]
```diff
@@ -717,6 +720,8 @@ public void ErrorsWithTheRequestStream_ShouldShutdownResponseStream()
 
         [Test]
         [Ignore("This test is flickering in our CI", Until = "2022-06-01 00:00:00Z")]
+        [NonParallelizable]
+        [Retry(10)]
```
[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:723]: I was able to make these fail locally as well when I had large test runs. I'd personally rather retry versus experience an intermittent failure again.. If it still consistently fails with 10 retries, then we can look deeper IMHO.

[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:723]: I've removed the retry attributes and am re-testing locally

[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:723]: It appears to work fine without the retry... pushing changes to see if we can build confidence with a 4th CI pass :)

File: tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs:758]: I think I actually see the core issue for this sleep being needed. The other one will need to stay though... testing a fix

[the remark this answers is not in the document]
Comment (JcolemanNR) [reply tests/Agent/UnitTests/Core.UnitTest/Spans/SpanStreamingServiceTests.cs]: I was able to get rid of THIS sleep, but I needed to add one somewhere else (with the retry attribute being removed).

**Pull request merged**